sql/grant.go

```diff
@@ -36,34 +35,25 @@ import (
 //   Notes: postgres requires the object owner.
 //          mysql requires the "grant option" and the same privileges, and sometimes superuser.
 func (p *planner) Grant(n *parser.Grant) (planNode, error) {
-	if len(n.Targets.Targets) == 0 {
-		return nil, errEmptyDatabaseName
-	}
-	if len(n.Targets.Targets) != 1 {
-		return nil, util.Errorf("TODO(marc): multiple targets not implemented")
-	}
-
-	// Lookup the database descriptor.
-	// TODO(marc): iterate over n.Targets.Targets once the grammar supports more than one.
-	dbDesc, err := p.getDatabaseDesc(n.Targets.Targets[0])
+	descriptor, err := p.getDescriptorFromTargetList(n.Targets)
 	if err != nil {
 		return nil, err
 	}
 
-	if !dbDesc.HasPrivilege(p.user, parser.PrivilegeWrite) {
+	if !descriptor.HasPrivilege(p.user, parser.PrivilegeWrite) {
 		return nil, fmt.Errorf("user %s does not have %s privilege on database %s",
-			p.user, parser.PrivilegeWrite, dbDesc.Name)
+			p.user, parser.PrivilegeWrite, descriptor.GetName())
 	}
 
-	if err := dbDesc.Grant(n); err != nil {
+	if err := descriptor.Grant(n); err != nil {
 		return nil, err
 	}
 
 	// Now update the descriptor.
 	// TODO(marc): do this inside a transaction. This will be needed
 	// when modifying multiple descriptors in the same op.
-	descKey := keys.MakeDescMetadataKey(dbDesc.ID)
-	if err := p.db.Put(descKey, dbDesc); err != nil {
+	descKey := keys.MakeDescMetadataKey(descriptor.GetID())
+	if err := p.db.Put(descKey, descriptor); err != nil {
```

Comment:
yeah. Although the descriptorProto interface is getting rather crowded. I'll may tweak Privilege to be an interface.